libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -189,21 +194,49 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
      * Updates annotations so that subsequent calls to getStatus will
      * reflect the most current annotations for the route.
      *
-     * @param route [DirectionsRoute]
+     * This methods manufactures a [DirectionsRefreshResponse] to adhere to requirements from
+     * https://github.com/mapbox/mapbox-navigation-native/pull/5420 where the full response has to be provided
+     * to [Navigator.updateAnnotations], not only the annotations/incidents collections.
      */
     override suspend fun updateAnnotations(route: NavigationRoute) {
-        route.directionsRoute.legs()?.forEachIndexed { index, routeLeg ->
+        val refreshedLegs = route.directionsRoute.legs()?.map { routeLeg ->
+            RouteLegRefresh.builder()
+                .annotation(routeLeg.annotation())
```

Comment:
Is it possible to add incidents here too?

Reply:
yes, it is. Can re-reuse the `udpateAnnotation` API for incidents? (not intuitive actually)